### DISCUSSION TITLE
List actual users, not clients connected to the network.

### DISCUSSION THREAD

**Discussion by samuelh2005** [Art-of-WiFi/UniFi-API-client#246] [Q&A] [answered]
Do anyone know is there a way to list actual users (eg the ones you find in **OS Settings** > **Admins & Users** > **Users**)?

There is `list_users()` but that calls `<controller_ip>/api/s/<site>/list/user` which returns a list of connected clients.

I believe that UniFi API client does not have this feature, but I am happy to program it myself if we can figure out the API endpoint.

**Comment by malle-pietje** [comment]
Not that I am aware. Where is this information available through the web interface?

**Reply by samuelh2005** [reply]
On the actual web interface it is located at `https://<controler_ip>/admins/users` and looks like this:

![Screenshot 2024-10-22 193750 - Copy](https://github.com/user-attachments/assets/3bf641ae-5fe5-4d77-a160-f0e868cf1c2a)

**Comment by malle-pietje** [comment]
Ah, now I understand what you’re looking for. Simply use `list_all_admins()` or `list_admins()`.

For questions like this we recommend using the API Browser tool to browse through the most popular collections:
https://github.com/Art-of-WiFi/UniFi-API-browser

**Reply by samuelh2005** [reply]
The problem with `list_all_admins()` or `list_admins()` is that it only returns users who have admin permisions (eg the users listed in the "Admins" tab in the image).

It doesn't return normal users, like "Bob Marly" in the image.

![Screenshot 2024-10-23 092613 - Copy](https://github.com/user-attachments/assets/894c9cf2-13eb-4f4f-bb25-cc5c0f28b127)

Here is what my "Admins" tab contains:

![Screenshot 2024-10-23 093018 - Copy](https://github.com/user-attachments/assets/afd9bce3-57c9-444a-8a4f-9be6ad0c9c9a)

as you can see there is no "Bob Marly", so he isn't returned by `list_all_admins()` or `list_admins()`.

**Reply by malle-pietje** [reply]
Ah, you are using Identity then? Don't believe that is supported through the API.

**Reply by malle-pietje** [reply]
I've looked closer at how this data is exposed on a test UDR we have in the lab it turns out there's an endpoint for this. Will try to add this `users` endpoint with the next release where we are adding the syslog entries as well. 🤞

**Reply by samuelh2005** [reply]
I am using Identity, but UniFi OS without Identity also supports creation of normal users, so I'm currently browsing through the machine with SSH to try and find where the users are stored.

**Reply by samuelh2005** [reply]
> I've looked closer at how this data is exposed on a test UDR we have in the lab it turns out there's an endpoint for this. Will try to add this `users` endpoint with the next release where we are adding the syslog entries as well. 🤞

That's good news, what is the endpoint so I can test it out?

**Reply by malle-pietje** [reply]
URL is: https://192.168.0.100/proxy/users/api/v2/users/
Method is: GET

It is going to be difficult to incorporate in this API client because it uses a different base URL.

**Reply by samuelh2005** [reply]
> URL is: https://192.168.0.100/proxy/users/api/v2/users/ Method is: GET
> 
> It is going to be difficult to incorporate in this API client because it uses a different base URL.

For me it looks like that API is invalid, if I open `https://10.0.0.1/proxy/users/api/v2/users/` with a web browser (which uses a GET request) I get redirected to `https://10.0.0.1/api/v2/users` and then given a `Not Found` error.

Maybe Identity is overiding the API with an unknown Identity API?

**Comment by malle-pietje** [comment, marked as answer]
Try this URL:
https://192.168.0.100/proxy/users/api/v2/users/search?including_resource=true&page_num=1&page_size=25&condition=&expand=with_last_activity,with_assignments&group_id=

**Reply by malle-pietje** [reply]
You need to be logged in with the same browser.

**Reply by samuelh2005** [reply]
Thanks, that's what I needed!
